src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -1341,6 +1341,7 @@ public PreferencesViewModel(DynamoViewModel dynamoViewModel)
             this.preferenceSettings.PropertyChanged += PreferenceSettings_PropertyChanged;
             this.pythonScriptEditorTextOptions = dynamoViewModel.PythonScriptEditorTextOptions;
             this.dynamoViewModel = dynamoViewModel;
+            this.dynamoViewModel.Model.ShutdownStarted += Model_ShutdownStarted;
```

Comment:
ensure that any event subscriptions you add are unsubscribed somewhere else, and make sure that place is actually called when it makes sense.